PR #2 for User Context Implementation - Rights Helper Interface

PhilMFischer opened the pull request:
This is no2 of the PR for multi user capabilities. This PR introduces a new Interface on the RightsHelper to clearly identify when the check for access is up to the system user or to the user context.

PR #571 needs to be merged before.

Commit:
Refactored Rights Helper Interface - (Task #548)

- Renamed old interface to SystemUserHasPermission to clearly identify
that this method is about the user that is signed up in the system.
- Adjusted all code to be compatible with this latest change

---
Task #548: Introduce capabilities to override current user for command
executions

Commit:
Merge remote-tracking branch 'origin/development' into feature/task_548-no2_Interface_Improvements

SaMuellerDLR commented on de.dlr.sc.virsat.model/src/de/dlr/sc/virsat/model/dvlm/roles/RightsHelper.java:
```diff
@@ -103,12 +103,8 @@ public static synchronized boolean hasWritePermission(EObject eObject, IUserCont
 	 * @return true in case the user has access or is a super user.
 	 */
-	public static boolean hasWritePermission(EObject eObject) {
+	//public static boolean hasSystemUserWritePermission(EObject eObject) {
```
Some leftover outcommented line.

PhilMFischer replied:
Removed...

SaMuellerDLR reviewed (changes requested):
Just a very minor thing.

Commit:
Fixes due to comments in review - (Task #548)

---
Task #548: Introduce capabilities to override current user for command
executions

SaMuellerDLR reviewed (approved):
Approved!
